projects/igniteui-angular/src/lib/grids/filtering/excel-style/grid.excel-style-filtering.component.ts

```diff
@@ -411,6 +412,15 @@ export class IgxGridExcelStyleFilteringComponent implements OnDestroy {
      * @hidden @internal
      */
     public onHideToggle() {
+        const eventArgs: IColumnVisibilityChangingEventArgs = {
+            column: this.column,
+            newValue: !this.column.hidden,
```

Comment:
the value should reamin as it were and change it only after cancelation is passed

Reply:
Please note that `newValue` is only passed in the event arguments and the value of `this.column.hidden` is not changed here.

Reply:
I see that you are using ! operator which changes it's state before the emition of the ing evt as you are doing after the canceletion which changes it's state IMO